docs/architecture/tuvali.md

```diff
@@ -0,0 +1,88 @@
+# Tuvali implementation
+Before we understand how Tuvali component performs activities on the BLE layer to transfer VC/VP. We can take a look at how BLE communication works in general between two devices A & B.
+
+## How does BLE Communication work?
+![BLE Communication](../_images/ble-high-level-comm-generic.png)
+
+In BLE communication, one device is designated as Peripheral and another one designated as Central.
+
+Peripheral can perform advertisement which can be read by a Central device and connect to it, if the advertisement is connectable.
+
+Once the connection is made, Central can perform further actions on the device like 
+- discovering services and characteristics
+- subscribing to notifications on characteritics
+- write/read data from characteristics
+- disconnect from device
+
+While peripheral can perform actions like 
+- sending notifications to subscribed characteristics
+- respond to read requests from Central
+
+The above diagram explains the sequence of actions for a BLE communciation in general.
+
+1. Advertisement from Peripheral
+2. Connection establishment & additional data exchange
```

Comment:
instead of `additional data exchange` we should mention as MTU negotiation?

Reply:
The highlighted section is explaining generic BLE communication but not the Tuvali implementation. 

However in the actual tuvali implementation (below this section), we have not mentioned MTU negotiation step, should we include that?